### ISSUE TITLE
Android 14: app not compatible 

### ISSUE THREAD

**Issue by rocketraman** [PhilippC/kp2a_accservice_autofill#45]
![Screenshot_20230706-013956.png](https://github.com/PhilippC/kp2a_accservice_autofill/assets/53049/846200a3-9e8c-42a5-87d5-8b38aedb4a41)

**Comment by DB1113** [comment]
Same Problem here.
Android 14 / Pixel 8 pro.

**Comment by elmicha** [comment]
I guess it's the same problem as in #41 .

**Comment by Utini2000** [comment]
Same issue here on Pixel 8 Pro :(

**Comment by Utini2000** [comment]
Seeing that @PhilippC hasn't posted/updated this for years: Does anyone know how to build it for 64-bit? The source code is available?

Otherwise this could be a major reason to move to another KeePass-related app.

**Comment by elmicha** [comment]
I tried to install it, too, and couldn't. But Autofill works just fine, without this plugin.

**Comment by Utini2000** [comment]
> I tried to install it, too, and couldn't. But Autofill works just fine, without this plugin.

Compared to my P6P (which has the plugin installed) I get a lot less "Fill in with KeePass2Android" buttons in apps and websites (using chrome).

**Comment by palexdev** [comment]
@PhilippC can you please update this?
Sometimes the Android auto fill doesn't work well and this offers a faster alternative to copy paste

**Comment by WebworkrNet** [comment]
Same here:
Samsung Galaxy A13 5G (SM-A136B)
Android 14, One UI-Version 6.0

**Comment by waterpolo** [comment]
Also on my Samsung Galaxy Tab S9 with A14 it couldn't install.

**Comment by waterpolo** [comment]
> I tried to install it, too, and couldn't. But Autofill works just fine, without this plugin.

Do you have root your device?
On my rooted Samsung Galaxy S23 with A14 autofill is working.

**Comment by ippocratis** [comment]
Using this  [apktool](https://github.com/rendiix/termux-apktool.git) on termux that has a  working aapt library changed the AndroidManifest.xml to include a

 `<uses-sdk android:minSdkVersion="21" android:targetSdkVersion="26" />`

And managed to install the apk 

tldr: the plugin does not work in A15 . Clicking on password fields that do no fire the autofil api also don't fire the accessibility service to use the kp2a app.

Anyway steps to install:

`apktool d autofill.apk -o decompiled_apk`

Edit AndroidManifest.xml under the decompiled_apk Dir 

`apktool b decompiled_apk -o fixed_autofill.apk`

`zipalign -v -p 4 fixed_autofill.apk aligned_autofill.apk`

`apksigner sign --ks /data/data/com.termux/files/home/keystore.jks --ks-key-alias my_alias --out signed_autofill.apk aligned_autofill.apk`

`apksigner verify signed_autofill.apk`

`su -c pm install signed_autofill.apk`

@PhilippC could you please consider updating kp2a since there are numerous issues with it in modern android versions .
App devs mostly don't care about autofill api so it is very common that it won't work in most apps .
This means even if its hacky the accessibility usage for autofill is essential.
Thanks

**Comment by kkuhle** [comment]
@PhilippC I agree completely. Autofill doesn't work on a frequent basis and it's really frustrating. I recently switched to Bitwarden and they have a setting to use accessibility to fill apps that don't support autofill and it makes things a lot better. I recently decided to explore KP2A accessibility again because it broke recently (they had an app redesign), but supposedly it will be fixed again in their next version. Anyway, I miss using keepass on PC and KP2A. If you could update this app to work, that would be amazing.

**Comment by waterpolo** [comment]
> [@PhilippC](https://github.com/PhilippC) I agree completely. Autofill doesn't work on a frequent basis and it's really frustrating. I recently switched to Bitwarden and they have a setting to use accessibility to fill apps that don't support autofill and it makes things a lot better. I recently decided to explore KP2A accessibility again because it broke recently (they had an app redesign), but supposedly it will be fixed again in their next version. Anyway, I miss using keepass on PC and KP2A. If you could update this app to work, that would be amazing.

Did you tried the recently pre release? I have a not rooted tablet which autofill in now working.

**Comment by kkuhle** [comment]
I did update to the latest pre-release (version 1.12-r3). The Autofill framework (keyboard) functions most of the time, but occasionally the accessibility option would be nice to have.
